docker/image.ts

```diff
@@ -75,32 +75,44 @@ export class Image extends pulumi.ComponentResource {
      */
     public imageName: pulumi.Output<string>;
     /**
-     * The built image digest.
+     * The built image Id.
      */
-    public digest: pulumi.Output<string>;
+    public id: pulumi.Output<string>;
+    /**
+     * The pushed image digest.
+     */
+    public digest: pulumi.Output<string | undefined>;
 
     constructor(name: string, args: ImageArgs, opts?: pulumi.ComponentResourceOptions) {
         super("docker:image:Image", name, args, opts);
 
-        const imageDigest = pulumi.output(args).apply(imageArgs => {
-            let localImageName = imageArgs.localImageName;
-            if (!localImageName) {
-                localImageName = imageArgs.imageName;
-            }
-            return buildAndPushImage(localImageName, imageArgs.build, imageArgs.imageName, this, async () => {
-                return {
-                    registry: imageArgs.registry.server,
-                    username: imageArgs.registry.username,
-                    password: imageArgs.registry.password,
-                };
-            });
-        });
+        const imageData =
+            pulumi
+            .all([args.imageName, args.build, args.localImageName, args.registry])
+            .apply(([imageName, build, localImageName, registry]) =>
+                pulumi
+                .all([registry.server, registry.username, registry.password])
+                .apply(async ([registryServer, username, password]) => {
```

Comment:
you undid this? :)

Reply:
I'll leave it undone for now so that we don't introduce a dependency on the newer `@pulumi/pulumi`.  But we should come back and redo this later.